extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -240,3 +248,117 @@ func RestoreRKE2K3SSnapshot(client *rancher.Client, clusterName string, snapshot
 
 	return nil
 }
+
+// RetentionLimitCheckRKE1 is a check that validates that the number of automatic snapshots on the cluster is under the retention limit
+func RetentionLimitCheckRKE1(client *rancher.Client, clusterName string) error {
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	if err != nil {
+		return err
+	}
+
+	clusterResp, err := client.Management.Cluster.ByID(clusterID)
+	if err != nil {
+		return err
+	}
+
+	retentionLimit := clusterResp.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.Retention
+
+	var s3Enabled bool = false
+	s3Config := clusterResp.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.S3BackupConfig
+	if s3Config != nil {
+		s3Enabled = true
+	}
+
+	existingSnapshots, err := GetRKE1Snapshots(client, clusterName)
+	automaticSnapshots := []management.EtcdBackup{}
+	for _, snapshot := range existingSnapshots {
+		if !snapshot.Manual {
+			automaticSnapshots = append(automaticSnapshots, snapshot)
+		}
+	}
```

Comment:
Great Idea I will  implement something for this. For Norman it looks like I can use Filters. I will hunt around for a way to do this in Steve.

Reply:
After looking into this a bit, i'm going to create a separate ticket for this since I would need to update all references to GETRKE1Snapshots to implement this (scope creep).